README.md
# www.iddpmirichmond.com
## Setup

###	Pre-requisites:
- Linux commandline knowledge
- Git Knowledge

### Mac Instructions

#### Mac Instructions: Install homebrew package manager:
- [https://brew.sh](https://brew.sh)

#### Mac Instructions: Install Python3

	$ brew install python3

#### Mac Instructions: Install Sublime Text 3
- Go to [https://www.sublimetext.com/3](https://www.sublimetext.com/3) and download the latest `OS X intaller`

### Windows Instructions

- What Do I Need?

	To effectively use Python on Windows, you need a few pieces of software:

	- Python
	- Cygwin
	- Sublime Text or another text editor

	Follow the instructions the below to set them up

	Also see for detailed reference: [https://www.davidbaumgold.com/tutorials/set-up-python-windows/](https://www.davidbaumgold.com/tutorials/set-up-python-windows/)

#### Windows Instructions: Download the latest python 3.x.x
- From [https://www.python.org/downloads/](https://www.python.org/downloads/) and install it. Use the `Windows x86-64 executable installer`. 
	- `For help reference instructions in` [https://www.davidbaumgold.com/tutorials/set-up-python-windows/](https://www.davidbaumgold.com/tutorials/set-up-python-windows/)

#### Windows Instructions: Install Cygwin
- From [https://cygwin.com/](https://cygwin.com/) download `setup.exe`, and run it.
	- Make sure to install the three packages mentioned on the instructions:
		- openssh
		- git
		- curl
			- `For help reference instructions in` [https://www.davidbaumgold.com/tutorials/set-up-python-windows/](https://www.davidbaumgold.com/tutorials/set-up-python-windows/)
	- Open Cygwin and test the installation of the packages:
		- ssh

			```
			$ ssh -V
			OpenSSH_8.2p1, OpenSSL 1.1.1d  10 Sep 2019
			```

		- git

			```
			$ git --version
			git version 2.21.0
			```
			- also configure you git email and username

				```
				$ git config --global user.email "you@example.com"
  				$ git config --global user.name "Your Name"
				```
				
		- curl

			```
			$ curl --version
			curl 7.66.0 (x86_64-pc-cygwin) ...
			...
			```
	- If you get any `command not found` error on the above commands it means you have to re-run the [https://cygwin.com/](https://cygwin.com/) download `setup.exe` and select to install the packages.

#### Windows Instructions: Install Sublime Text 3
- Go to [https://www.sublimetext.com/3](https://www.sublimetext.com/3) and download the latest `Windows 64 bit intaller`
	- `For help reference instructions in` [https://www.davidbaumgold.com/tutorials/set-up-python-windows/](https://www.davidbaumgold.com/tutorials/set-up-python-windows/)


#### Windows Instructions: Test your python installation
- `For help reference instructions in` [https://www.davidbaumgold.com/tutorials/set-up-python-windows/](https://www.davidbaumgold.com/tutorials/set-up-python-windows/)

	Run the commands below to test your python installation

	```
	python -c 'print("Hello, World!")'
	Hello, World!
	``` 
- If that works, lets move on to writing a program into a file. Open up Sublime Text, and type in the following:
	
	```
	#!/usr/bin/env python
	print("Hello, World!")
	```
	
	Save the file in your home directory (inside of C:\cygwin\home) as hello.py. 

	Then open up the Terminal, cd to that directory, and run:
	
	```
	$ chmod a+x hello.py
	$ ./hello.py
	Hello, World!
	```

	And you should see the same Hello, World! output that you saw before. 
	
	Congratulations, you're ready to start learning Python!

### Set Github ssh key
- Follow the instructions on: [https://help.github.com/en/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent](https://help.github.com/en/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)

### Create the directories to store our github source code

```
$ cd
$ mkdir -p source/github.com/iddpmirichmondca/
```

### Dowload the church-website repo 

```
$ cd  source/github.com/iddpmirichmondca/
$ git clone git@github.com:iddpmirichmondca/church-website.git
$ git clone git@github.com:iddpmirichmondca/church-website.git
Cloning into 'church-website'...
The authenticity of host 'github.com (140.82.113.4)' can't be established.
RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
Warning: Permanently added 'github.com,140.82.113.4' (RSA) to the list of known hosts.
remote: Enumerating objects: 66, done.
remote: Counting objects: 100% (66/66), done.
remote: Compressing objects: 100% (50/50), done.
remote: Total 66 (delta 14), reused 53 (delta 7), pack-reused 0
Receiving objects: 100% (66/66), 192.41 KiB | 791.00 KiB/s, done.
Resolving deltas: 100% (14/14), done.
```

### Create a Python3 Virtual Environment inside church-website repo

- change directory to the church-website repo

```	
$ cd church-website
```

- Create python virtual environment for the project
	- Mac

		```
		$ python3 -m venv .venv
		```

		This will create the .venv directory if it doesn’t exist, and also create directories inside it containing a copy of the Python interpreter, the standard library, and various supporting files.

		- Activate the virtual environment

			```
			$ source .venv/bin/activate
			```

		- Install all required packages

			```
			(.venv) $ pip install -r requirements.txt
			```

		- Run website locally

			```
			(.venv) $ cd mysite/
			(.venv) $ python manage.py migrate
   			(.venv) $ python manage.py runserver
   			```
	- Windows

		You'll need to open a Windows PowerShell in order tu run your python virtual environment.

		```
		PS set-executionpolicy remotesigned 
		PS cd C:\cygwin64\home\colon\source\github.com\iddpmirichmondca\church-website\
		PS .\.venv\Scripts\Activate.ps1
		(.venv) PS
		pip install -r .\requirements.txt
		```
		WIP


### Access your local copy
- **Website:** [http://localhost:8000](http://localhost:8000)
	- Website: 
![website.png](docs/images/website.png)
- **Admin:** [http://localhost:8000/admin](http://localhost:8000/admin)
	- Admin sign in:
![admin-login.png](docs/images/admin-login.png)
	- Admin panel
![admin-panel.png](docs/images/admin-panel.png) 

### Shutdown local running website
- Press `control` + `c` key to shutdown local server

CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [1.0.0]	-	2019-02-15

### Added

- Initial app development





<!-- # Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [1.1.0]	- 2019-02-15

### Added

- Changelog inconsistency section in Bad Practices

## [1.0.0] - 2017-06-20

### Added

- New visual identity by [@tylerfortune8](https://github.com/tylerfortune8).
- Version navigation.
- Links to latest released version in previous versions.
- "Why keep a changelog?" section.
- "Who needs a changelog?" section.
- "How do I make a changelog?" section.
- "Frequently Asked Questions" section.
- New "Guiding Principles" sub-section to "How do I make a changelog?".
- Simplified and Traditional Chinese translations from [@tianshuo](https://github.com/tianshuo).
- German translation from [@mpbzh](https://github.com/mpbzh) & [@Art4](https://github.com/Art4).
- Italian translation from [@azkidenz](https://github.com/azkidenz).
- Swedish translation from [@magol](https://github.com/magol).
- Turkish translation from [@karalamalar](https://github.com/karalamalar).
- French translation from [@zapashcanon](https://github.com/zapashcanon).
- Brazilian Portugese translation from [@Webysther](https://github.com/Webysther).
- Polish translation from [@amielucha](https://github.com/amielucha) & [@m-aciek](https://github.com/m-aciek).
- Russian translation from [@aishek](https://github.com/aishek).
- Czech translation from [@h4vry](https://github.com/h4vry).
- Slovak translation from [@jkostolansky](https://github.com/jkostolansky).
- Korean translation from [@pierceh89](https://github.com/pierceh89).
- Croatian translation from [@porx](https://github.com/porx).
- Persian translation from [@Hameds](https://github.com/Hameds).
- Ukrainian translation from [@osadchyi-s](https://github.com/osadchyi-s).

### Changed

- Start using "changelog" over "change log" since it's the common usage.
- Start versioning based on the current English version at 0.3.0 to help
  translation authors keep things up-to-date.
- Rewrite "What makes unicorns cry?" section.
- Rewrite "Ignoring Deprecations" sub-section to clarify the ideal
  scenario.
- Improve "Commit log diffs" sub-section to further argument against
  them.
- Merge "Why can’t people just use a git log diff?" with "Commit log
  diffs"
- Fix typos in Simplified Chinese and Traditional Chinese translations.
- Fix typos in Brazilian Portuguese translation.
- Fix typos in Turkish translation.
- Fix typos in Czech translation.
- Fix typos in Swedish translation.
- Improve phrasing in French translation.
- Fix phrasing and spelling in German translation.

### Removed

- Section about "changelog" vs "CHANGELOG".

## [0.3.0] - 2015-12-03

### Added

- RU translation from [@aishek](https://github.com/aishek).
- pt-BR translation from [@tallesl](https://github.com/tallesl).
- es-ES translation from [@ZeliosAriex](https://github.com/ZeliosAriex).

## [0.2.0] - 2015-10-06

### Changed

- Remove exclusionary mentions of "open source" since this project can
  benefit both "open" and "closed" source projects equally.

## [0.1.0] - 2015-10-06

### Added

- Answer "Should you ever rewrite a change log?".

### Changed

- Improve argument against commit logs.
- Start following [SemVer](https://semver.org) properly.

## [0.0.8] - 2015-02-17

### Changed

- Update year to match in every README example.
- Reluctantly stop making fun of Brits only, since most of the world
  writes dates in a strange way.

### Fixed

- Fix typos in recent README changes.
- Update outdated unreleased diff link.

## [0.0.7] - 2015-02-16

### Added

- Link, and make it obvious that date format is ISO 8601.

### Changed

- Clarified the section on "Is there a standard change log format?".

### Fixed

- Fix Markdown links to tag comparison URL with footnote-style links.

## [0.0.6] - 2014-12-12

### Added

- README section on "yanked" releases.

## [0.0.5] - 2014-08-09

### Added

- Markdown links to version tags on release headings.
- Unreleased section to gather unreleased changes and encourage note
  keeping prior to releases.

## [0.0.4] - 2014-08-09

### Added

- Better explanation of the difference between the file ("CHANGELOG")
  and its function "the change log".

### Changed

- Refer to a "change log" instead of a "CHANGELOG" throughout the site
  to differentiate between the file and the purpose of the file — the
  logging of changes.

### Removed

- Remove empty sections from CHANGELOG, they occupy too much space and
  create too much noise in the file. People will have to assume that the
  missing sections were intentionally left out because they contained no
  notable changes.

## [0.0.3] - 2014-08-09

### Added

- "Why should I care?" section mentioning The Changelog podcast.

## [0.0.2] - 2014-07-10

### Added

- Explanation of the recommended reverse chronological release ordering.

## [0.0.1] - 2014-05-31

### Added

- This CHANGELOG file to hopefully serve as an evolving example of a
  standardized open source project CHANGELOG.
- CNAME file to enable GitHub Pages custom domain
- README now contains answers to common questions about CHANGELOGs
- Good examples and basic guidelines, including proper date formatting.
- Counter-examples: "What makes unicorns cry?"

[unreleased]: https://github.com/olivierlacan/keep-a-changelog/compare/v1.1.0...HEAD
[1.1.0]: https://github.com/olivierlacan/keep-a-changelog/compare/v1.0.0...v1.1.0
[1.0.0]: https://github.com/olivierlacan/keep-a-changelog/compare/v0.3.0...v1.0.0

mysite/mysite/settings/base.py
"""
Django settings for mysite project.

Generated by 'django-admin startproject' using Django 2.2.10.

For more information on this file, see
https://docs.djangoproject.com/en/2.2/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/2.2/ref/settings/
"""

# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
import os
from django.utils.translation import gettext_lazy as _

PROJECT_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
BASE_DIR = os.path.dirname(PROJECT_DIR)


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/2.2/howto/deployment/checklist/


# Application definition

INSTALLED_APPS = [
    # This project
    'website',

    # CodeRed CMS
    'coderedcms',
    'bootstrap4',
    'modelcluster',
    'taggit',
    'wagtailfontawesome',
    'wagtailcache',
    'wagtailimportexport',

    # Wagtail
    'wagtail.contrib.forms',
    'wagtail.contrib.redirects',
    'wagtail.embeds',
    'wagtail.sites',
    'wagtail.users',
    'wagtail.snippets',
    'wagtail.documents',
    'wagtail.images',
    'wagtail.search',
    'wagtail.core',
    'wagtail.contrib.settings',
    'wagtail.contrib.modeladmin',
    'wagtail.contrib.table_block',
    'wagtail.admin',

    # Django
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'django.contrib.sitemaps',
]

MIDDLEWARE = [
    # Save pages to cache. Must be FIRST.
    'wagtailcache.cache.UpdateCacheMiddleware',

    # Common functionality
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.common.CommonMiddleware',

    # Security
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
    'django.middleware.security.SecurityMiddleware',

    #  Error reporting. Uncomment this to recieve emails when a 404 is triggered.
    # 'django.middleware.common.BrokenLinkEmailsMiddleware',

    # CMS functionality
    'wagtail.core.middleware.SiteMiddleware',
    'wagtail.contrib.redirects.middleware.RedirectMiddleware',

    # Fetch from cache. Must be LAST.
    'wagtailcache.cache.FetchFromCacheMiddleware',
]

ROOT_URLCONF = 'mysite.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
                'wagtail.contrib.settings.context_processors.settings',
            ],
        },
    },
]

WSGI_APPLICATION = 'mysite.wsgi.application'


# Database
# https://docs.djangoproject.com/en/2.2/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
    }
}


# Password validation
# https://docs.djangoproject.com/en/2.2/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]

# Internationalization
# https://docs.djangoproject.com/en/2.2/topics/i18n/

# To add or change language of the project, modify the list below.
LANGUAGE_CODE = 'en-us'

LANGUAGES = [
    ('en-us', _('English'))
]

TIME_ZONE = 'America/New_York'

USE_I18N = True

USE_L10N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/2.2/howto/static-files/

STATICFILES_FINDERS = [
    'django.contrib.staticfiles.finders.FileSystemFinder',
    'django.contrib.staticfiles.finders.AppDirectoriesFinder',
]

STATIC_ROOT = os.path.join(BASE_DIR, 'static')
STATIC_URL = '/static/'

MEDIA_ROOT = os.path.join(BASE_DIR, 'media')
MEDIA_URL = '/media/'


# Login

LOGIN_URL = 'wagtailadmin_login'
LOGIN_REDIRECT_URL = 'wagtailadmin_home'


# Wagtail settings

WAGTAIL_SITE_NAME = 'IDDPMI Richmond CA'

WAGTAIL_ENABLE_UPDATE_CHECK = False

# Base URL to use when referring to full URLs within the Wagtail admin backend -
# e.g. in notification emails. Don't include '/admin' or a trailing slash
BASE_URL = 'http://www.iddpmirichmond.com'


# Bootstrap

BOOTSTRAP4 = {
    # set to blank since coderedcms already loads jquery and bootstrap
    'jquery_url': '',
    'base_url': '',
    # remove green highlight on inputs
    'success_css_class': ''
}


# Tags

TAGGIT_CASE_INSENSITIVE = True


mysite/website/migrations/0002_initial_data.py

# -*- coding: utf-8 -*-
from __future__ import unicode_literals

from django import VERSION as DJANGO_VERSION
from django.db import migrations


def initial_data(apps, schema_editor):
    ContentType = apps.get_model('contenttypes.ContentType')
    Page = apps.get_model('wagtailcore.Page')
    Site = apps.get_model('wagtailcore.Site')
    WebPage = apps.get_model('website.WebPage')

    # Create page content type
    webpage_content_type, created = ContentType.objects.get_or_create(
        model='webpage',
        app_label='website',
    )

    # Delete the default home page generated by wagtail,
    # and replace it with a more useful page type.
    curr_homepage = Page.objects.filter(slug='home').delete()

    homepage = WebPage.objects.create(
        title = "Home",
        slug='home',
        custom_template='coderedcms/pages/home_page.html',
        content_type=webpage_content_type,
        path='00010001',
        depth=2,
        numchild=0,
        url_path='/home/',
    )

    # Create a new default site
    Site.objects.create(
        hostname='www.iddpmirichmond.com',
        site_name='IDDPMI Richmond CA',
        root_page_id=homepage.id,
        is_default_site=True
    )


class Migration(migrations.Migration):

    dependencies = [
        ('coderedcms', '0001_initial'),
        ('wagtailcore', '0002_initial_data'),
        ('website', '0001_initial'),
    ]

    operations = [
        migrations.RunPython(initial_data),
    ]


mysite/mysite/settings/prod.py
from .base import *  # noqa

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = False

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'sx%-$t8=_zaak*8e_b5nsg(u%&8zi&rx622j*w59rg4(s9hog1'

# Add your site's domain name(s) here.
ALLOWED_HOSTS = ['www.iddpmirichmond.com']

# To send email from the server, we recommend django_sendmail_backend
# Or specify your own email backend such as an SMTP server.
# https://docs.djangoproject.com/en/2.2/ref/settings/#email-backend
EMAIL_BACKEND = 'django_sendmail_backend.backends.EmailBackend'

# Default email address used to send messages from the website.
DEFAULT_FROM_EMAIL = 'IDDPMI Richmond CA <info@iddpmirichmond.com>'

# A list of people who get error notifications.
ADMINS = [
    ('Administrator', 'admin@iddpmirichmond.com'),
]

# A list in the same format as ADMINS that specifies who should get broken link
# (404) notifications when BrokenLinkEmailsMiddleware is enabled.
MANAGERS = ADMINS

# Email address used to send error messages to ADMINS.
SERVER_EMAIL = DEFAULT_FROM_EMAIL

# DATABASES = {
#     'default': {
#         'ENGINE': 'django.db.backends.mysql',
#         'HOST': 'localhost',
#         'NAME': 'mysite',
#         'USER': 'mysite',
#         'PASSWORD': '',
#         # If using SSL to connect to a cloud mysql database, spedify the CA as so.
#         'OPTIONS': { 'ssl': { 'ca': '/path/to/certificate-authority.pem' } },
#     }
# }

# Use template caching to speed up wagtail admin and front-end.
# Requires reloading web server to pick up template changes.
TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
                'wagtail.contrib.settings.context_processors.settings',
            ],
            'loaders': [
                ('django.template.loaders.cached.Loader', [
                    'django.template.loaders.filesystem.Loader',
                    'django.template.loaders.app_directories.Loader',
                ]),
            ],
        },
    },
]

CACHES = {
    'default': {
        'BACKEND': 'django.core.cache.backends.filebased.FileBasedCache',
        'LOCATION': os.path.join(BASE_DIR, 'cache'),  # noqa
        'KEY_PREFIX': 'coderedcms',
        'TIMEOUT': 14400,  # in seconds
    }
}

try:
    from .local_settings import *  # noqa
except ImportError:
    pass
